### DISCUSSION TITLE
Does MuJoCo support returning states as graphs?

### DISCUSSION THREAD

**Discussion by davidireland-iso** [google-deepmind/mujoco#287] [Asking for Help]
I am thinking of e.g. the work done in [NerveNet](https://www.cs.toronto.edu/~tingwuwang/nervenet.html) where the bodies of the agents are broken down into e.g. limbs and modelled using a graph, where, for instance each limb would be a node and an edge would connected limbs depending on when they touch. 

Do any MuJoCo environments support an option where you can choose to have the state returned in such a way, or is it still required to implement this ourselves?

**Comment by nimrod-gileadi** [comment]
There is no such support in the MuJoCo API, which we try to keep quite low level, though of course MuJoCo's way of representing kinematic chains exactly corresponds to such a graph.
